twitter_timeline/main.py

```diff
@@ -22,25 +24,149 @@ def before_request():
     g.db = connect_db(settings.DATABASE_NAME)
 
 
+# Views go here:
 @app.route('/friendship', methods=['POST', 'DELETE'])
 @json_only
 @auth_only
 def friendship(user_id):
-    pass
+    '''
+    Creates or deletes a "following relationship" between an authenticated
+    user and another user whose username is sent in a POST json object.
+    '''
+    
+    # Verify a username was sent
+    try:
+        followed_username = request.json['username']
+    except:
+        abort(400)
+    
+    # Verify that user exists
+    followed_user = g.db.users.find_one({'username': followed_username})
+    if not followed_user:
+        abort(400)
+    
+    # Get the follower object from the users collection
+    follower = g.db.users.find_one({'_id': ObjectId(user_id)})
```

Comment:
I think the `user_id` is already of type `ObjectId` (when it is coming from the decorator) so the `ObjectId()` conversion here is redundant (also on line 82)